packages/amazonq/src/app/inline/completion.ts

```diff
@@ -52,71 +49,167 @@ export const CodewhispererInlineCompletionLanguages = [
+            const params: LogInlineCompletionSessionResultsParams = {
+                sessionId: sessionId,
+                completionSessionResult: {
+                    [itemId]: {
+                        seen: true,
+                        accepted: true,
+                        discarded: false,
+                    },
                 },
-            },
-            totalSessionDisplayTime: Date.now() - requestStartTime,
-            firstCompletionDisplayLatency: firstCompletionDisplayLatency,
+                totalSessionDisplayTime: Date.now() - requestStartTime,
+                firstCompletionDisplayLatency: firstCompletionDisplayLatency,
+            }
+            this.languageClient.sendNotification(this.logSessionResultMessageName, params)
+            this.disposable.dispose()
+            this.disposable = languages.registerInlineCompletionItemProvider(
+                CodewhispererInlineCompletionLanguages,
+                this.inlineCompletionProvider
+            )
+        }
+        commands.registerCommand('aws.amazonq.acceptInline', onInlineAcceptance)
+
+        const onInlineRejection = async () => {
+            await commands.executeCommand('editor.action.inlineSuggest.hide')
+            // TODO: also log the seen state for other suggestions in session
+            this.disposable.dispose()
+            this.disposable = languages.registerInlineCompletionItemProvider(
+                CodewhispererInlineCompletionLanguages,
+                this.inlineCompletionProvider
+            )
+            const sessionId = this.sessionManager.getActiveSession()?.sessionId
+            const itemId = this.sessionManager.getActiveRecommendation()[0]?.itemId
+            if (!sessionId || !itemId) {
+                return
+            }
+            const params: LogInlineCompletionSessionResultsParams = {
+                sessionId: sessionId,
+                completionSessionResult: {
+                    [itemId]: {
+                        seen: true,
+                        accepted: false,
+                        discarded: false,
+                    },
+                },
+            }
+            this.languageClient.sendNotification(this.logSessionResultMessageName, params)
         }
-        languageClient.sendNotification(logInlineCompletionSessionResultsNotificationType as any, params)
+        commands.registerCommand('aws.amazonq.rejectCodeSuggestion', onInlineRejection)
+
+        /*
+            We have to overwrite the prev. and next. commands because the inlineCompletionProvider only contained the current item
+            To show prev. and next. recommendation we need to re-register a new provider with the previous or next item
+        */
+
+        const swapProviderAndShow = async () => {
+            await commands.executeCommand('editor.action.inlineSuggest.hide')
+            this.disposable.dispose()
+            this.disposable = languages.registerInlineCompletionItemProvider(
```

Comment:
Does this still have the weird effect where the number of items shown in the inline completion don't get updated? I guess thats unavoidable though?

Reply:
yes it still has that issue and it's not avoidable unfortunately